Reuse the endpoint of the checkpointed container when restore

huikang opened the pull request:
Docker-DCO-1.1-Signed-off-by: Hui Kang hkang.sunysb@gmail.com

Commit:
Checkpoint/Restore Support: add exec driver methods

Methods for checkpointing and restoring containers were added to the
native driver.  The LXC driver returns an error message that these
methods are not implemented yet.

Signed-off-by: Saied Kazemi <saied@google.com>

Conflicts:
	daemon/execdriver/native/create.go
	daemon/execdriver/native/driver.go
	daemon/execdriver/native/init.go

Commit:
Checkpoint/Restore Support: add exec driver methods

Methods for checkpointing and restoring containers were added to the
native driver.  The LXC driver returns an error message that these
methods are not implemented yet.

Signed-off-by: Saied Kazemi <saied@google.com>

Conflicts:
	daemon/execdriver/native/create.go
	daemon/execdriver/native/driver.go
	daemon/execdriver/native/init.go

Commit:
Checkpoint/Restore Support: add exec driver methods

Methods for checkpointing and restoring containers were added to the
native driver.  The LXC driver returns an error message that these
methods are not implemented yet.

Signed-off-by: Saied Kazemi <saied@google.com>

Conflicts:
	daemon/execdriver/native/create.go
	daemon/execdriver/native/driver.go
	daemon/execdriver/native/init.go

Commit:
Checkpoint/Restore Support: add functionality to daemon

Support was added to the daemon to use the Checkpoint and Restore methods
of the native exec driver for checkpointing and restoring containers.

Signed-off-by: Saied Kazemi <saied@google.com>

Conflicts:
	api/server/server.go
	daemon/container.go
	daemon/daemon.go
	daemon/networkdriver/bridge/driver.go
	daemon/state.go
	vendor/src/github.com/docker/libnetwork/ipallocator/allocator.go

Commit:
Checkpoint/Restore Support: add functionality to daemon

Support was added to the daemon to use the Checkpoint and Restore methods
of the native exec driver for checkpointing and restoring containers.

Signed-off-by: Saied Kazemi <saied@google.com>

Conflicts:
	api/server/server.go
	daemon/container.go
	daemon/daemon.go
	daemon/networkdriver/bridge/driver.go
	daemon/state.go
	vendor/src/github.com/docker/libnetwork/ipallocator/allocator.go

Commit:
Checkpoint/Restore Support: add functionality to daemon

Support was added to the daemon to use the Checkpoint and Restore methods
of the native exec driver for checkpointing and restoring containers.

Signed-off-by: Saied Kazemi <saied@google.com>

Conflicts:
	api/server/server.go
	daemon/container.go
	daemon/daemon.go
	daemon/networkdriver/bridge/driver.go
	daemon/state.go
	vendor/src/github.com/docker/libnetwork/ipallocator/allocator.go

huikang committed:
Release the network resource during checkpoint

Restore failed if network resource not released during checkpoint,
e.g., a container with port open with -p

Signed-off-by: Hui Kang <hkang.sunysb@gmail.com>

Conflicts:
	daemon/container.go

huikang committed:
Release the network resource during checkpoint

Restore failed if network resource not released during checkpoint,
e.g., a container with port open with -p

Signed-off-by: Hui Kang <hkang.sunysb@gmail.com>

Conflicts:
	daemon/container.go

huikang committed:
Release the network resource during checkpoint

Restore failed if network resource not released during checkpoint,
e.g., a container with port open with -p

Signed-off-by: Hui Kang <hkang.sunysb@gmail.com>

Conflicts:
	daemon/container.go

boucher committed:
Update checkpoint/restore support to match docker/master

Docker-DCO-1.1-Signed-off-by: Ross Boucher <rboucher@gmail.com> (github: boucher)

boucher committed:
Update checkpoint/restore support to match docker/master

Docker-DCO-1.1-Signed-off-by: Ross Boucher <rboucher@gmail.com> (github: boucher)

boucher committed:
Update checkpoint/restore support to match docker/master

Docker-DCO-1.1-Signed-off-by: Ross Boucher <rboucher@gmail.com> (github: boucher)

boucher committed:
Update daemon and cli support for checkpoint and restore.

Docker-DCO-1.1-Signed-off-by: Ross Boucher <rboucher@gmail.com> (github: boucher)

boucher committed:
Update daemon and cli support for checkpoint and restore.

Docker-DCO-1.1-Signed-off-by: Ross Boucher <rboucher@gmail.com> (github: boucher)

boucher committed:
Update daemon and cli support for checkpoint and restore.

Docker-DCO-1.1-Signed-off-by: Ross Boucher <rboucher@gmail.com> (github: boucher)

boucher committed:
Add compilation steps for Criu to the Dockerfile
Add a basic test for checkpoint/restore to the integration tests

Docker-DCO-1.1-Signed-off-by: Ross Boucher <rboucher@gmail.com> (github: boucher)

boucher committed:
Add compilation steps for Criu to the Dockerfile
Add a basic test for checkpoint/restore to the integration tests

Docker-DCO-1.1-Signed-off-by: Ross Boucher <rboucher@gmail.com> (github: boucher)

boucher committed:
Add compilation steps for Criu to the Dockerfile
Add a basic test for checkpoint/restore to the integration tests

Docker-DCO-1.1-Signed-off-by: Ross Boucher <rboucher@gmail.com> (github: boucher)

boucher committed:
Add optional dependency info to the PACKAGERS file.

Docker-DCO-1.1-Signed-off-by: Ross Boucher <rboucher@gmail.com> (github: boucher)

boucher committed:
Add optional dependency info to the PACKAGERS file.

Docker-DCO-1.1-Signed-off-by: Ross Boucher <rboucher@gmail.com> (github: boucher)

boucher committed:
Add optional dependency info to the PACKAGERS file.

Docker-DCO-1.1-Signed-off-by: Ross Boucher <rboucher@gmail.com> (github: boucher)

boucher committed:
Don't destroy/delete the container if it has been checkpointed.

Docker-DCO-1.1-Signed-off-by: Ross Boucher <rboucher@gmail.com> (github: boucher)

boucher committed:
Don't destroy/delete the container if it has been checkpointed.

Docker-DCO-1.1-Signed-off-by: Ross Boucher <rboucher@gmail.com> (github: boucher)

boucher committed:
Don't destroy/delete the container if it has been checkpointed.

Docker-DCO-1.1-Signed-off-by: Ross Boucher <rboucher@gmail.com> (github: boucher)

huikang committed:
Reuse the endpoint of the checkpointed container when restore

Docker-DCO-1.1-Signed-off-by: Hui Kang <hkang.sunysb@gmail.com>

huikang commented:
@boucher This PR fixes the network endpoint problem when restoring a container. The endpoint should be reused by a checkpointed container so that the network resources (e.g., iptable rule) will not be changed after restoring. Please review if it fits into your branch. Thanks.

boucher commented:
If I run this code, and then checkpoint and restore a container, it still can't access the network. I believe this is related to the issue I filed here: https://github.com/docker/docker/issues/14027

(To test, I just exec'd into the restored container and tried ping'ing google.com) Are you actually able to use the network after a restore?

huikang committed:
Reuse endpoint for 1.8.0-dev

Docker-DCO-1.1-Signed-off-by: Hui Kang <hkang.sunysb@gmail.com>

Merge branch 'cr-combined' into cr-combined-ep-fix

Conflicts:
	daemon/container.go
	daemon/container_linux.go

huikang commented:
@boucher Update the patch for 1.8.0-dev, but the problem you described still exists. I will look into this issue these days.

boucher committed:
Move checkpoint methods into a separate container_checkpoint file.

Docker-DCO-1.1-Signed-off-by: Ross Boucher <rboucher@gmail.com> (github: boucher)

huikang committed:
Merge remote-tracking branch 'upstream/cr-combined' into cr-combined

huikang committed:
Update endpoint fix code

Docker-DCO-1.1-Signed-off-by: Hui Kang <hkang.sunysb@gmail.com> (github: huikang)
